packages/core/test/smoke-valibot.test.ts

```diff
@@ -629,3 +629,115 @@ describe("extending presets", () => {
     });
   });
 });
+
+describe("createFinalSchema", () => {
+  test("custom schema combiner", () => {
+    let receivedIsServer = false;
+    const env = createEnv({
+      server: {
+        SERVER_ENV: v.string(),
+      },
+      shared: {
+        SHARED_ENV: v.string(),
+      },
+      clientPrefix: "CLIENT_",
+      client: {
+        CLIENT_ENV: v.string(),
+      },
+      runtimeEnv: {
+        SERVER_ENV: "server",
+        SHARED_ENV: "shared",
+        CLIENT_ENV: "client",
+      },
+      createFinalSchema: (shape, isServer) => {
+        expectTypeOf(isServer).toEqualTypeOf<boolean>();
+        if (typeof isServer === "boolean") receivedIsServer = true;
```

Comment:
OOC, why not just `expect(isServer).toBe(true)` here?

Reply:
if createFinalSchema wasn't called you'd never get an error, since the expect call never happened

Reply:
Ah right